File: TorcLibrary/LibraryData/Entities/Book.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LibraryRepository.Entities
{
    public class Book
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Column("Book_Id")]
        public int Id { get; set; }

        [Required]
        public string Title { get; set; }

        [Column("First_Name")]
        public string FirstName { get; set; }

        [Column("Last_Name")]
        public string LastName { get; set; }

        [Column("Total_Copies")]
        public int TotalCopies { get; set; }

        public string Type { get; set; }
        public string ISBN { get; set; }
        public string Category { get; set; }
        public string Status { get; set; }
    }
}

File: ui/src/types/Book.ts
export interface Book {
  id: number;
  title: string;
  firstName: string;
  lastName: string;
  totalCopies: number;
  type: string;
  isbn: string;
  category: string;
  status: string;
}

export interface BookSearchParams {
  author?: string;
  isbn?: string;
  ownershipStatus?: string;
  offset?: number;
  limit?: number;
}

export interface BookSearchResult {
  books: Book[];
  totalCount: number;
  hasMore: boolean;
}

export type SearchFilterType = 'author' | 'isbn' | 'status';